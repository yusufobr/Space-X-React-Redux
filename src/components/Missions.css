table,
th,
td {
  border: 1px solid lightgray;
  border-collapse: collapse;
}

th {
  padding-left: 16px;
}
