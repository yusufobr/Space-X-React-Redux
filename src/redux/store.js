// eslint-disable-nextline
import { configureStore } from '@reduxjs/toolkit';

const store = configureStore({
  reducer: {

  },
});

export default store;
