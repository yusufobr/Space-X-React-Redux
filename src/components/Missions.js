import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { addReserve } from '../redux/missions/missionsSlice';
import './Missions.css';

function Missions() {
  const { missions } = useSelector((state) => state.missions);
  const dispatch = useDispatch();

  return (
    <div className="container mx-auto">
      <table className="text-left">
        <thead>
          <tr>
            <th>Mission</th>
            <th>Description</th>
            <th>Status</th>
          </tr>
        </thead>
        <tbody>
          {missions.map((mission, index) => (
            <tr
              key={mission.mission_id}
              className={index % 2 === 0 ? 'bg-gray-100' : ''}
            >
              <td className="text-lg font-bold w-[12.5%] p-4">
                {mission.mission_name}
              </td>
              <td className="p-3 pb-8">{mission.description}</td>
              <td className="w-[12.5%] text-center">
                <span
                  className={
                    mission.reserved
                      ? 'bg-[#18A2B8] text-white font-semibold uppercase px-2 rounded-md'
                      : 'bg-gray-500 text-white font-semibold uppercase px-2 rounded-md'
                  }
                >
                  {mission.reserved ? 'active member' : 'not a member'}
                </span>
              </td>
              <td className="w-[12.5%] text-center">
                <button
                  type="button"
                  className={
                    mission.reserved
                      ? 'py-1 px-3 rounded border-2 text-red-500 border-red-500'
                      : 'py-1 px-3 rounded border-2 border-black'
                  }
                  id={mission.mission_id}
                  onClick={() => {
                    dispatch(addReserve(mission.mission_id));
                  }}
                >
                  {mission.reserved ? 'Leave Mission' : 'Join Mission'}
                </button>
              </td>
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  );
}

export default Missions;
