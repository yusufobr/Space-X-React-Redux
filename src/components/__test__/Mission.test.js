import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../redux/store';
import Missions from '../Missions';

describe('Test Mission Component', () => {
  it('Should renders Missions component without errors', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Missions />
      </Provider>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders a list of missions', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Missions />
        </Provider>,
      )
      .toJSON();
    expect(tree.children.length).toBe(1);
  });
});
