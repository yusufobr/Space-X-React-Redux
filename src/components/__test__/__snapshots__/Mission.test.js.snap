// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Mission Component Should renders Missions component without errors 1`] = `
<div
  className="container mt-8 px-4 mx-auto"
>
  <p
    className="text-left text-2xl"
  >
    loading ...
  </p>
</div>
`;
